src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,104 @@
+package duke;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Scanner;
+import java.io.*;
+
+import duke.task.*;
+import duke.command.TaskParser;
+
+public class Duke {
+
+    static ArrayList<Task> taskList = new ArrayList<>();
+    static final String FILENAME = "tasklist";
+
+    private static void saveTaskListToFile() {
+	try {
+	    FileOutputStream fileos = new FileOutputStream(FILENAME);
+	    ObjectOutputStream objos = new ObjectOutputStream(fileos);
+	    objos.writeObject(taskList);
+	} catch (IOException e) {
+	    System.out.println("Error saving to file.");
+	    System.out.println(e.getMessage());
+	}
+    }
+
+    private static void readTaskListFromFile() {
+	try {
+	    FileInputStream fileis = new FileInputStream(FILENAME);
+	    ObjectInputStream objis = new ObjectInputStream(fileis);
+	    taskList = (ArrayList<Task>) objis.readObject();
+	} catch (IOException | ClassNotFoundException e) {
+	    System.out.println("Error reading from file.");
+	    System.out.println(e.getMessage());
+	}
+    }
+
+    private static boolean handleOneInputLine(String line) {
+        String[] splitted = line.split("\\s+");
+        switch (splitted[0]) {
+            case "bye":
```

Comment:
String constants like `"bye"`, `"list"` should be added as named constants, for example: `BYE_COMMAND = "bye"`